src/pages/index.js

```diff
@@ -6,46 +6,83 @@ import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import ProfilePicEvt from "../components/ProfilePicEvt.js";
 import {
-  initialCards,
   config,
   editProfileButton,
+  editProfilePictureButton,
   addCardButton,
+  profilePicture,
 } from "../utils/constants.js";
+import { api } from "../components/Api.js";
+let gallery;
 //Add Card Popup Form
 const addCardPopupForm = new PopupWithForm("#add-card-modal", {
   handleSubmitEvent: (inputValues) => {
-    const newCard = {
-      name: inputValues.name,
-      link: inputValues.bio,
-    };
-    const cardElement = createCard(newCard);
-    gallery.addItem(`prepend`, cardElement);
+    addCardPopupForm.renderLoading(true);
+    api
+      .addCard(inputValues)
+      .then((inputValues) => {
+        const cardElement = createCard(inputValues);
+        gallery.addItem(`prepend`, cardElement);
+        addCardPopupForm.renderLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        addCardPopupForm.renderLoading(false);
+      });
     addCardPopupForm.form.reset();
   },
 });
 // Edit Popup Form
 const editPopupForm = new PopupWithForm("#edit-profile-modal", {
   handleSubmitEvent: (inputValues) => {
-    userInfo.setUserInfo({ name: inputValues.name, job: inputValues.bio });
+    editPopupForm.renderLoading(true);
+    api
+      .updateUserInfo(inputValues)
+      .then((inputValues) => {
+        userInfo.setUserInfo({
+          name: inputValues.name,
+          job: inputValues.about,
+        });
+        editPopupForm.renderLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        editPopupForm.renderLoading(false);
+      });
   },
 });
 const nameInput = editPopupForm.form.querySelector('input[name="name"]');
 const jobInput = editPopupForm.form.querySelector('input[name="bio"]');
-const userInfo = new UserInfo(".profile__name", ".profile__bio");
-// Image Review Popup
-const popupWithImage = new PopupWithImage("#open-image-modal");
-// Image Gallery
-const gallery = new Section(
+const userInfo = new UserInfo(
+  ".profile__name",
+  ".profile__bio",
+  ".profile__picture__image"
+);
+// Edit profile popup form
+const editProfilePictureForm = new PopupWithForm(
+  "#edit-profile-picture-modal",
   {
-    data: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      gallery.addItem(`append`, cardElement);
+    handleSubmitEvent: (inputValues) => {
+      api
+        .updateUserPic(inputValues)
+        .then((data) => {
+          userInfo.setUserPic({ avatar: data.avatar });
+          editProfilePictureForm.renderLoading(true);
+        })
+        .catch((err) => {
+          console.log(err);
+          editProfilePictureForm.renderLoading(false);
```

Comment:
* `....renderLoading(false)` should be in `finally`, not to duplicate it in `then` and `catch`